scripts/analyze_benchmarks/extractor.ts

```diff
@@ -0,0 +1,88 @@
+import { readFile } from 'fs/promises';
+
+export type BenchmarkEntry = {
+    instructions: { __bigint__: string };
+    method_name: string;
+    timestamp: { __bigint__: string };
+};
+
+type CanisterBenchmark = {
+    current?: {
+        version: string;
+        benchmarks: BenchmarkEntry[];
+    };
+    previous: {
+        version: string;
+        benchmarks: BenchmarkEntry[];
+    };
+};
+
+type BenchmarksJson = {
+    [canisterName: string]: CanisterBenchmark;
+};
+
+/**
+ * Extracts benchmark entries from multiple files and groups them by version
+ * @param files Array of file paths to process
+ * @returns A record mapping version strings to arrays of benchmark entries
+ */
+export async function extractBenchmarksEntriesFromFiles(
+    files: string[]
+): Promise<Record<string, BenchmarkEntry[]>> {
+    const versionEntriesArrays = await Promise.all(
+        files.map(extractBenchmarkEntries)
+    );
+    const versionEntries = versionEntriesArrays.flat();
+
+    return groupEntriesByVersion(versionEntries);
+}
+
+/**
+ * Extracts benchmark entries from a single file
+ * @param file Path to the benchmark file
+ * @returns Array of tuples containing version and benchmark entry pairs
+ */
+async function extractBenchmarkEntries(
```

Comment:
This function could be declaratized more. There are multiple levels of nesting of lambdas

Reply:
done